go/src/lib.rs

```diff
@@ -227,35 +224,36 @@ pub unsafe extern "C" fn free_connection_response(
 /// * `free_command_response` can only be called once per `CommandResponse`. Calling it twice is undefined behavior, since the address will be freed twice.
 /// * `command_response_ptr` must be obtained from the `CommandResponse` returned in [`SuccessCallback`] from [`command`].
 /// * `command_response_ptr` must be valid until `free_command_response` is called.
-/// * The contained `string_value` must be obtained from the `CommandResponse` returned in [`SuccessCallback`] from [`command`].
-/// * The contained `string_value` must be valid until `free_command_response` is called and it must outlive the `CommandResponse` that contains it.
-/// * The contained `array_elements_len` must be obtained from the `CommandResponse` returned in [`SuccessCallback`] from [`command`].
-/// * The contained `array_elements_len` must be valid until `free_command_response` is called and it must outlive the `CommandResponse` that contains it.
-/// * The contained `array_value` must be obtained from the `CommandResponse` returned in [`SuccessCallback`] from [`command`].
-/// * The contained `array_value` must be valid until `free_command_response` is called and it must outlive the `CommandResponse` that contains it.
 #[no_mangle]
 pub unsafe extern "C" fn free_command_response(command_response_ptr: *mut CommandResponse) {
     assert!(!command_response_ptr.is_null());
     let command_response = unsafe { Box::from_raw(command_response_ptr) };
+    free_command_response_elements(*command_response);
+}
+
+/// Frees the nested elements of `CommandResponse`.
+///
+/// # Safety
+///
+/// * `free_command_response_elements` can only be called once per `CommandResponse`. Calling it twice is undefined behavior, since the address will be freed twice.
+/// * The contained `string_value` must be obtained from the `CommandResponse` returned in [`SuccessCallback`] from [`command`].
+/// * The contained `string_value` must be valid until `free_command_response` is called and it must outlive the `CommandResponse` that contains it.
+/// * The contained `array_value` must be obtained from the `CommandResponse` returned in [`SuccessCallback`] from [`command`].
+/// * The contained `array_value` must be valid until `free_command_response` is called and it must outlive the `CommandResponse` that contains it.
+fn free_command_response_elements(command_response: CommandResponse) {
```

Comment:
we should add a test for checking for memory leak here

Reply:
I have added a TODO for now to address this in the upcoming PR as the findings could take time and would block upcoming PRs as well.